### PR TITLE
mgmt, local generation for Microsoft.Web 2023-01-01

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#4753](https://github.com/Azure/sdk-release-request/issues/4753)

# Description

1. Add `revapi` configuration in pom.xml because this time `ManagedServiceIdentity` and `UserAssignedIdentity` have been changed to final class.
2. Add non-encryption judgment on `StorageAccountKey` in test cases `FunctionAppsTests#canCRUDFunctionApp` and `FunctionAppsTests#canCRUDLinuxFunctionApp`.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java [hunk 1]
```diff
@@ -76,6 +77,7 @@ protected void cleanUpResources() {
     }
 
     @Test
+    @DoNotRecord(skipInPlayback = true)
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java:80]: why?

File: sdk/resourcemanager/azure-resourcemanager-appservice/pom.xml [hunk 2]
```diff
@@ -51,6 +51,7 @@
 
       --add-opens com.azure.core/com.azure.core.implementation.util=ALL-UNNAMED
     </javaModulesSurefireArgLine>
+    <revapi.skip>true</revapi.skip>
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/pom.xml:54]: why?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-appservice/pom.xml:54]: The test cases `canCRUDFunctionApp` and `canCRUDLinuxFunctionApp` can work in `RECORD` and `LIVE` modes, but not in `PLAYBACK` mode. The detail infomation following as URL: 
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3267132&view=results

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java:80]: The test cases `canCRUDFunctionApp` and `canCRUDLinuxFunctionApp` can work in `RECORD` and `LIVE` modes, but not working in `PLAYBACK` mode (the local machine too). The detail infomation following as URL:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3267132&view=results

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java:80]: Not a valid reason. CI works before with the Playback. Please check with Xiaofei on the recording.

File: sdk/resourcemanager/azure-resourcemanager-appservice/pom.xml (hunk 2 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-appservice/pom.xml:54]: This time `ManagedServiceIdentity` and `UserAssignedIdentity` have been changed to final class.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java [hunk 3]
```diff
@@ -104,9 +102,12 @@ public void canCRUDFunctionApp() throws Exception {
                 functionAppResource1.appSettings.get(KEY_AZURE_WEB_JOBS_STORAGE).value(),
                 functionAppResource1.appSettings.get(KEY_CONTENT_AZURE_FILE_CONNECTION_STRING).value());
         // verify accountKey
-        Assertions
-            .assertEquals(
-                functionAppResource1.storageAccount.getKeys().get(0).value(), functionAppResource1.accountKey);
+        String accountKey = functionAppResource1.storageAccount.getKeys().get(0).value();
+        if (!"REDACTED".equals(accountKey)) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java:106]: if (!isPlaybackMode()) {

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/FunctionAppsTests.java]: This judgment is fixed to `if (!isPlaybackMode())`

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**